### DISCUSSION TITLE
Choice of GUI: TkInter vs. PyQt?

### DISCUSSION THREAD

**Discussion by pwdirks** [MorseKOB/PyKOB#207] [General]
Hi guys,

I remember discussion about adding a proper 'Preferences' panel to set various config options in the GUI and I thought I might look into that over the holiday break.

I recall some discussions about possibly switching from using TkInter to PyQT - did that ever get settled?  Pros/Cons of each?  It'd be a shame to add a bunch of TkInter GUI only to add to the burden of a port to PyQT if we should decide that's really the way we want to be going (in which case perhaps my energy might be better spent on a port to PyQT in the first place).

Thoughts?

**Comment by leskerr** [comment]
Hi @pwdirks! A 'Preferences' panel is sorely needed, and it would be wonderful if you could look into it.

As for the choice of GUI, I expect my role will transition to more of a user and less of an implementer going forward, and I don't know anything about PyQT, so my vote doesn't count for much. The reason I went with TkInter in the first place is because it appeared to be the standard GUI for Python and I wanted something that would be familiar to any Python programmers who might want to dive into the MKOB code.

**Comment by jchausler** [comment]
Hi Les and Pat,  Patrick, I've just done a "Reply", not a reply all so please tell me whether you get that funny data within the response.  And I'm using IE 11.I'm not a GUI Guy so have no idea whether TkInter or PyQT would be the better choice.  The only thing I can suggest is that I would think that package which provides some combination of the best of performance, flexibility, ease of programming and wide platform (MAC, Linux. Windows) support should be the one of choice.  Also as I mentioned some time ago, I've wondered whether that issue with delayed sound might be dealt with by using some compiled version of Python and the applicability of the above packages to such and environment.73, Chris-----Original Message-----

**Comment by pwdirks** [comment]
Hi Chris,

If by “that funny data” you mean HTML for a “DIV” in an HTML comment, then yes.

Thanks for your points - all valid.  TkInter is supplied as part of the Python distribution so it’s certainly widely supported but I believe PyQT is also supported on Mac, Windows, and Linux.  I don’t recall whether QT or TkInter had the more “platform-like” GUI.  IIRC Ed had looked at both and had some opinions.  I’ll wait for him to comment, too.

73, N6PWD
-Patrick.

**Comment by jchausler** [comment]
Hi All,Well, since just reply and reply all seem both to cause the problem, maybe I need to send such information to some other email address.  What would that be?73, Chris-----Original Message-----

**Reply by AESilky** [reply]
Chris, another problem is that your reply is including the content you are replying to. That causes the content to be repeated over and over within the issue. I think you should try the `view it on GitHub` link at the bottom of the email and add your comment withing the issue.

**Comment by AESilky** [comment]
Chris,
Yes, it still showed up. Another option is to use the link in the email
'Open in GitHub'. Then you can see all of the previous comments and enter
yours.  I generally only use the 'reply' when I'm on my phone (like now).

I'll comment on Tkinter and QT this evening.

**Comment by AESilky** [comment]
In the discussion page I opened for discussion about Tkinter vs PyQt. I included a link to a pros/cons page. I have looked at a few comparisons, but I've never actually used PyQt. I've actually done very little work with Tkinter. The majority of my Python programming has been with ROS (Robot Operating System) to program my robots and autonomous car (1/10 scale Mini Cooper) which don't have a GUI. So I don't know enough to suggest one over the other.

The whole GUI needs to be one or the other - you can't easily mix them, so I will agree with @pwdirks that if we want to switch to PyQt the initial work would be to convert the existing UI.

I don't think it would be a small task to convert, because (my understanding is that) the two aren't really similar (except for concepts that pretty much all GUI frameworks/libraries have in common).

To answer @jchausler, they are both cross platform across Windows, Mac, and Linux. My understanding is that PyQt provides a GUI that is more 'platform correct' while Tkinter provides one that looks more similar across the platforms - but I can't confirm or deny that.

**Comment by AESilky** [comment]
I'm marking this as CLOSED, since there is so much TKinter in MKOB that it would be hard to switch it. I will note that 'Telegram' (the only other app with a GUI) uses PyGame (the community supported version, not the original) since its interface requirements are significantly different from MKOB. MRT (and the utilities) use a CLI, so this discussion doesn't apply.
